warehouse/ingest-core/src/main/java/datawave/ingest/mapreduce/partition/SortedTabletLocationPartitioner.java

```diff
@@ -0,0 +1,100 @@
+package datawave.ingest.mapreduce.partition;
+
+import com.google.common.collect.TreeMultimap;
+import datawave.ingest.mapreduce.job.SplitsFileType;
+import org.apache.hadoop.io.Text;
+import org.apache.log4j.Logger;
+
+import java.util.*;
+
+public class SortedTabletLocationPartitioner extends MultiTableRangePartitioner {
+
+    private static final Logger log = Logger.getLogger(SortedTabletLocationPartitioner.class);
+    private final Map<String,Map<Integer,Integer>> SPLIT_TO_REDUCER_MAP = new HashMap<>();
+
+    @Override
+    protected int calculateIndex(int index, int numPartitions, String tableName, int cutPointArrayLength) {
+        if(isAssignedPartition(tableName, index)){
+
+        }
+        assignPartitions(numPartitions, tableName, cutPointArrayLength);
+        if (isAssignedPartition(tableName, index)) {
+            return SPLIT_TO_REDUCER_MAP.get(tableName).get(index);
+        }
+        log.trace("Index was not found after recomputing the reducer map");
+        return 0;
+    }
+
+    private void assignPartitions(int numPartitions, String tableName, int cutPointArrayLength) {
+        Map<Text, String> currentTableSplitToLocation = splitToLocationMap.get().get(tableName);
+        Map<Integer,Integer> tempSplitReducerMap = new HashMap<>();
+        Text[] cutPointArray = splitsByTable.get().get(tableName);
+
+        if (cutPointArrayLength > numPartitions) {
+            mapPartitions(numPartitions, cutPointArrayLength, currentTableSplitToLocation, tempSplitReducerMap, cutPointArray);
```

Comment:
should this result also be stored in the SPLIT_TO_REDUCER_MAP?

Reply:
I'm not sure I understand the comment.  Could you elaborate?

Reply:
They are added to tempSplitReducerMap in the mapPartitions method which is then added to SPLIT_TO_REDUCER map.